--- nemerle-0.2.1/configure	2004-09-13 18:08:56.000000000 +0200
+++ nemerle-0.2.1/configure.new	2004-09-28 14:21:03.598384235 +0200
@@ -46,6 +46,7 @@
 dll_path=
 tmp_dir=
 nem_tmp_file=
+use_ngen=1
 
 nem_flags=
 time=
@@ -178,6 +179,8 @@
                         Mono .NET environment or if you don't use Mono
                         at all but have it installed
   --ignore-errors       do not exit on errors
+  --disable-aot         do not try to mono --aot/ngen the assemblies
+  --enable-aot          try to mono --aot/ngen the assemblies (default)
 
 Optional features:
   --enable-debug        adds --debug to net-flags
@@ -238,6 +241,12 @@
         --disable-quiet)
 	    quiet=
 	    ;;
+	--disable-aot)
+	    use_ngen=0
+	    ;;
+	--enable-aot)
+	    use_ngen=1
+	    ;;
 	--ignore-errors)
 	    ignore_errors=yes
 	    ;;
@@ -377,20 +386,27 @@
 # Step 5.
 # Check how to invoke native image generator.
 echo_check_for "native image generator"
-try_execute "ngen.exe"
-if test "$?" != 127 -a "$?" != 126; then
-    echo_result "ngen.exe"
-    ngen="ngen"
+
+if test "$use_ngen" = 0 ; then
+    echo_result disabled
+    ngen="true"
 else
-    try_execute "mono --aot"
+    try_execute "ngen.exe"
     if test "$?" != 127 -a "$?" != 126; then
-	echo_result "mono --aot"
-	ngen="mono --aot"
+        echo_result "ngen.exe"
+        ngen="ngen"
     else
-	echo_result "not found"
-	echo "*** Please make sure that the native image generator is ***"
-	echo "*** available as a single executable file, somewhere in PATH. ***"
-	abort "Haven't found any native image generators on your system."
+        try_execute "mono --aot"
+        if test "$?" != 127 -a "$?" != 126; then
+    	    echo_result "mono --aot"
+    	    ngen="mono --aot"
+        else
+    	    echo_result "not found"
+    	    echo "*** Please make sure that the native image generator is ***"
+    	    echo "*** available as a single executable file, somewhere in PATH. ***"
+	    ngen="true"
+	    use_ngen=0
+        fi
     fi
 fi
 
@@ -439,7 +455,11 @@
     gacutil_base="gacutil -root \$(DESTDIR)$mono_libdir"
     gacutil_command="$gacutil_base -check_refs -package nemerle -i"
     gacutil_uninstall_command="$gacutil_base -u"
-    ngen_gac="for f in \$(DESTDIR)$mono_libdir/mono/gac/Nemerle*/*/*.dll ; do mono --aot \$\$f ; done"
+    if [ "$use_ngen" = "0" ] ; then
+      ngen_gac="true"
+    else
+      ngen_gac="for f in \$(DESTDIR)$mono_libdir/mono/gac/Nemerle*/*/*.dll ; do mono --aot \$\$f ; done"
+    fi
   else
     echo_result no
     gacutil_command="gacutil /f /i"
